### PR TITLE
Cheatcodes docs

### PR DESCRIPTION
**Pull request by kasperski95**
Added asserts and cheatcodes docs.

### CODE REVIEW

File: website/docs/tutorials/guides/testing.md [hunk 1]
```diff
@@ -2,10 +2,185 @@
 
 T.B.D.
 
+## Asserts
+
+Protostar ships with own assert functions. They don't accept [implicit arguments](https://www.cairo-lang.org/docs/how_cairo_works/builtins.html?highlight=implicit%20arguments#implicit-arguments) compared to asserts from [`starkware.cairo.common.math`](https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/cairo/common/math.cairo). You can import Protostar asserts in the following way:
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/testing.md:7]:
```suggestion
Protostar ships with its own assert functions. They don't accept [implicit arguments](https://www.cairo-lang.org/docs/how_cairo_works/builtins.html?highlight=implicit%20arguments#implicit-arguments) compared to asserts from [`starkware.cairo.common.math`](https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/cairo/common/math.cairo). You can import Protostar asserts in the following way:
```

File: website/docs/tutorials/guides/testing.md [hunk 2]
```diff
@@ -2,10 +2,185 @@
...
+`$(which protostar)/../cairo` to the [`CAIRO_PATH`](https://www.cairo-lang.org/docs/how_cairo_works/imports.html?highlight=cairo_path).
+:::
+
+You can find all [assert signatures here](https://github.com/software-mansion/protostar/blob/master/cairo/protostar/asserts.cairo).
+
 ## Cheatcodes
 
+Most of the time, testing smart contracts with assertions only is not enough. Some test cases require manipulating the state of the blockchain, as well as checking for reverts and events. For that reason, Protostar provides a set of cheatcodes.
+
+Cheatcodes are available in [hints](https://www.cairo-lang.org/docs/hello_cairo/program_input.html#hints). You don't have to import anything. When Protostar runs tests, it replaces some core elements in [cairo-lang](https://pypi.org/project/cairo-lang/) library and inject cheatcodes to the hint scope.
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/testing.md:27]:
```suggestion
Cheatcodes are available inside of [Cairo hints](https://www.cairo-lang.org/docs/hello_cairo/program_input.html#hints). You don't have to import anything. When Protostar runs tests, it replaces some core elements in [cairo-lang](https://pypi.org/project/cairo-lang/) library and inject cheatcodes to the hint scope.
```

File: website/docs/tutorials/guides/testing.md [hunk 3]
```diff
@@ -2,10 +2,185 @@
...
+
+To use `mock_call` effectively, you need to understand how Cairo data structures are represented under the hood. `Cairo-lang` operates on list of integers. The following examples demonstrate how each data structure is represented in Python code.
+
+```cairo title="Fixtures and type definitions"
+const EXTERNAL_CONTRACT_ADDRESS = 0x3fe90a1958bb8468fb1b62970747d8a00c435ef96cda708ae8de3d07f1bb56b
+
+struct Point:
+    member x : felt
+    member y : felt
+end
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/testing.md:51]: Can we move it into the Struct section? Would be easier to comprehend

File: website/docs/tutorials/guides/testing.md [hunk 4]
```diff
@@ -2,10 +2,185 @@
...
+namespace ITestContract:
+    func get_felt() -> (res : felt):
+    end
+
+    func get_array() -> (res_len : felt, res : felt*):
+    end
+
+    func get_struct() -> (res : Point):
+    end
+end
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/testing.md:63]: Maybe we can split it into three small contract, each for one example? would be easier to comprehend

File: website/docs/tutorials/guides/testing.md [hunk 5]
```diff
@@ -2,10 +2,185 @@
...
+
+Removes a mocked call specified by a function name (`fn_name`) of a contract with an address (`contract_address`).
+
+### `expect_revert`
+
+```python
+def expect_revert() -> None: ...
+```
+
+If a code beneath `expect_revert` triggers revert, a test will pass.
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/testing.md:132]: What will happen if it won't trigger a revert?

File: website/docs/tutorials/guides/testing.md [hunk 6]
```diff
@@ -2,10 +2,185 @@
...
+    BasicContract.increase_balance(contract_address=contract_a_address, amount=3)
+    return ()
+end
+```
+
+### `deploy_contract`
+
+```python
+def deploy_contract(contract_path: str) -> Contract:
+```
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/testing.md:152]: Can we link the related section here?

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]
I think it's very good, some minor nitpicks in comments. 

I have a one broader question though, can we make a section about mapping `python -> cairo` more feature intependent? We will need to link it for constructor arguments in the near future too. If you have some simple idea how to do it I would appreciate it, otherwise we can postpone it for now.

File: website/docs/tutorials/guides/testing.md (hunk 6 above)
Comment (kasperski95) [reply website/docs/tutorials/guides/testing.md:152]: Did you merge your testing docs to master?

Comment (kasperski95) [reply website/docs/tutorials/guides/testing.md:152]: I'll just copy what you did in that PR. I tried to pull the branch, but there are too many changes.

**Comment by kasperski95** [comment]
> I have a one broader question though, can we make a section about mapping python -> cairo more feature intependent? We will need to link it for constructor arguments in the near future too. If you have some simple idea how to do it I would appreciate it, otherwise we can postpone it for now.

A problem is that a `mock_call` expects an array for a single felt. Let's not create a wrong "abstraction".

File: website/docs/tutorials/guides/testing.md (hunk 6 above)
Comment (MaksymilianDemitraszek) [reply website/docs/tutorials/guides/testing.md]: If you copy integration testing here, could you please move it on the top of the list? I think this cheat code is the most important one

**Review by MaksymilianDemitraszek — APPROVED** [review]
LGTM

**Pull request merged**